framework/Steps.py

```diff
@@ -427,16 +434,34 @@ def _localInitializeStep(self,inDictionary):
     if inDictionary['Model'].createWorkingDir:
       currentWorkingDirectory = os.path.join(inDictionary['jobHandler'].runInfoDict['WorkingDir'],
                                              inDictionary['jobHandler'].runInfoDict['stepName'])
-      try:
-        os.mkdir(currentWorkingDirectory)
-      except FileExistsError:
-        self.raiseAWarning('current working dir '+currentWorkingDirectory+' already exists, ' +
-                           'this might imply deletion of present files')
-        if utils.checkIfPathAreAccessedByAnotherProgram(currentWorkingDirectory,3.0):
-          self.raiseAWarning('directory '+ currentWorkingDirectory + ' is likely used by another program!!! ')
-        if utils.checkIfLockedRavenFileIsPresent(currentWorkingDirectory,self.lockedFileName):
-          self.raiseAnError(RuntimeError, self, "another instance of RAVEN is running in the working directory "+ currentWorkingDirectory+". Please check your input!")
-        # register function to remove the locked file at the end of execution
+      workingDirReady = False
+      alreadyTried = False
+      while not workingDirReady:
+        try:
+          os.mkdir(currentWorkingDirectory)
+          workingDirReady = True
+        except FileExistsError:
+          if self._clearRunDir and not alreadyTried:
+            self.raiseAWarning(f'The calculation run directory {currentWorkingDirectory} already exists, ' +
+                              'clearing existing files. This action can be disabled through the RAVEN Step input.')
+
+            utils.removeDir(currentWorkingDirectory)
+            alreadyTried = True
+            continue
+          else:
+            if alreadyTried:
+              self.raiseAWarning(f'The calculation run directory {currentWorkingDirectory} already exists, ' +
+                                'and was not able to be removed. ' +
+                                'This might imply replacing files present, or unintended error handling.')
+            else:
+              self.raiseAWarning(f'The calculation run directory {currentWorkingDirectory} already exists, ' +
+                                'this might imply replacing files present, or unintended error handling.')
+            if utils.checkIfPathAreAccessedByAnotherProgram(currentWorkingDirectory,3.0):
```

Comment:
I would test this before actually trying to remove the folder... it might be possibile that for 3 sec or in that particular moment the underlying code is not using/writing/accessing files and we might delete the folder before checking for the ravenLocked file and/or the last access time.

Reply:
Okay, sure. I am moving them.